File: 1492. The kth Factor of n/main.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        int num = 1;

        while (num <= n) {
            if (n % num == 0) {
                k--;
                if (k == 0) {
                    return num;
                }
            }

            num++;
        }

        return -1;
    }
};

class Solution {
public:
    int kthFactor(int n, int k) {
        int count = 0;

        for (int i = 1; i <= n; i++) {
            if (n % i == 0) {
                count++;
                if (count == k) {
                    return i;
                }
            }
        }

        return -1;
    }
};

File: 1492. The kth Factor of n/main.rs
struct Solution;

impl Solution {
    pub fn kth_factor(n: i32, k: i32) -> i32 {
        let mut num = 1;
        let mut k = k;

        while num <= n {
            if n % num == 0 {
                k -= 1;
                if k == 0 {
                    return num;
                }
            }

            num += 1;
        }

        -1
    }
}

impl Solution {
    pub fn kth_factor(n: i32, k: i32) -> i32 {
        let mut count = 0;

        for i in 1..=n {
            if n % i == 0 {
                count += 1;
                if count == k {
                    return i;
                }
            }
        }

        -1
    }
}

impl Solution {
    pub fn kth_factor(n: i32, k: i32) -> i32 {
        let mut fact = Vec::new();

        for i in 1..=n {
            if n % i == 0 {
                fact.push
            }
        }

        -1
    }
}
